mne/tests/test_epochs.py

```diff
@@ -724,6 +727,29 @@ def test_epochs_proj():
     data_2 = epochs_read.get_data()  # Let's check the result
     assert_allclose(data, data_2, atol=1e-15, rtol=1e-3)
 
+    # adding EEG ref (GH #2727)
+    raw = Raw(raw_fname)
+    raw.add_proj([], remove_existing=True)
+    raw.info['bads'] = ['MEG 2443', 'EEG 053']
+    picks = pick_types(raw.info, meg=False, eeg=True, stim=True, eog=False,
+                       exclude='bads')
+    epochs = Epochs(raw, events, event_id, tmin, tmax, proj=True, picks=picks,
+                    baseline=(None, 0), preload=True, add_eeg_ref=False)
+    epochs.pick_channels(['EEG 001', 'EEG 002'])
+    assert_equal(len(epochs), 7)  # sufficient for testing
+    temp_fname = 'test.fif'
+    epochs.save(temp_fname)
+    epochs = read_epochs(temp_fname, add_eeg_ref=True, proj=True)
+    assert_allclose(epochs.get_data().mean(axis=1), 0, atol=1e-15)
+    epochs = read_epochs(temp_fname, add_eeg_ref=True, proj=False)
+    assert_raises(AssertionError, assert_allclose,
+                  epochs.get_data().mean(axis=1), 0., atol=1e-15)
```

Comment:
It should be that projs are applied during `get_data()` if `proj=True` regardless of preload state. Preload state should never change what you get from `get_data`, if it does there is a bug with our preloading logic. I'll add a test though to be sure.